scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/CascadingBackend.scala

```diff
@@ -391,31 +392,29 @@ object CascadingBackend {
    * step before building the Flow. Job also needs to call this
    * method in validate to make sure validation works.
    */
-  def planTypedWrites(fd: FlowDef, mode: Mode): Unit =
-    FlowStateMap.mutate(fd) { st =>
-      val writes = st.pendingTypedWrites
-      val empty = Dag.empty(OptimizationRules.toLiteral)
-      type ToDo[A] = (Id[A], TypedSink[A])
-      val (rootedDag, todos) = writes.foldLeft((empty, List.empty[ToDo[_]])) { case ((dag, items), tw) =>
-        val (nextDag, id) = dag.addRoot(tw.pipe)
-        require(tw.mode == mode, s"${tw.mode} should be equal to $mode")
-        (nextDag, (id, tw.sink) :: items)
-      }
-      val phases = defaultOptimizationRules(
-        mode match {
-          case h: HadoopMode => Config.fromHadoop(h.jobConf)
-          case _ => Config.empty
-        })
-      val optDag = rootedDag.applySeq(phases)
-      def doWrite[A](pair: (Id[A], TypedSink[A])): Unit = {
-        val optPipe = optDag.evaluate(pair._1)
-        val dest = pair._2
-        val cpipe = toPipeUnoptimized[A](optPipe, dest.sinkFields)(fd, mode, dest.setter)
-        dest.writeFrom(cpipe)(fd, mode)
-      }
-      todos.foreach(doWrite(_))
-      (st.copy(pendingTypedWrites = Nil), ())
+  def planTypedWrites(fd: FlowDef, mode: Mode): Unit = {
+    val writes = FlowStateMap.removeWrites(fd).pendingTypedWrites
```

Comment:
nit: very confusing signature of `removeWrites`, without knowledge that `removeWrites` return original `flowState` is hard to understand this line.